server/src/test/java/org/elasticsearch/cluster/coordination/AtomicRegisterCoordinatorTests.java

```diff
@@ -514,29 +513,15 @@ public boolean shouldJoinLeaderInTerm(long currentTerm, long targetTerm) {
         }
 
         @Override
-        public void onNewElection(
-            DiscoveryNode localNode,
-            long proposedTerm,
-            ClusterState latestAcceptedState,
-            ActionListener<Void> listener
-        ) {
+        public void onNewElection(DiscoveryNode localNode, long proposedTerm, ActionListener<Void> listener) {
             ActionListener.completeWith(listener, () -> {
-                final var latestClusterState = sharedStore.getLatestClusterState();
-                if (latestClusterState != null && latestClusterState.version() > latestAcceptedState.version()) {
-                    throw new CoordinationStateRejectedException("The node has an stale applied cluster state version");
-                }
-
-                final var latestAcceptedClusterUUID = latestAcceptedState.metadata().clusterUUID();
-                final var proposedNewTermOwner = new TermOwner(localNode, proposedTerm, latestAcceptedClusterUUID);
+                final var proposedNewTermOwner = new TermOwner(localNode, proposedTerm);
                 final var witness = register.claimTerm(proposedNewTermOwner);
                 if (proposedNewTermOwner != witness) {
-                    if (witness.clusterUUID().equals(latestAcceptedClusterUUID) == false) {
-                        assert latestAcceptedState.metadata().clusterUUIDCommitted() == false;
-                        atomicRegisterPersistedState.changeAcceptedClusterUUID(witness.clusterUUID());
-                    }
                     throw new CoordinationStateRejectedException("Term " + proposedTerm + " already claimed by another node");
                 }
                 lastWonTerm = proposedTerm;
+                atomicRegisterPersistedState.updateLatestAcceptedStateIfStale();
```

Comment:
I wonder if we should allow `onNewElection` to return the cluster state for the coordinator to accept properly? Otherwise we'll need to make the same change in the production implementation at some point.

Reply:
Sorry this was bad advice, see https://github.com/elastic/elasticsearch/pull/94497#discussion_r1143336303 for a better idea.

Reply:
No problem 👍